spec/index.md

```diff
@@ -721,23 +729,74 @@ In order to make a query call to canister, the user makes a POST request to `/ap
 
 -   `arg` (`blob`): Argument to pass to the canister method.
 
-If the call resulted in a reply, the response is a CBOR (see [CBOR](#cbor)) map with the following fields:
+Canister methods that do not change the canister state (except for cycle balance change due to message execution) can be executed more efficiently. This method provides that ability, and returns the canister's response directly within the HTTP response.
+
+If the query call resulted in a reply, the response is a CBOR (see [CBOR](#cbor)) map with the following fields:
 
--   `status` (`text`): `replied`
+-   `status` (`text`): `"replied"`
 
 -   `reply`: a CBOR map with the field `arg` (`blob`) which contains the reply data.
 
+-   (optional) `signatures` (`[+ node-signature]`): a list containing one node signature for the returned query response.
```

Comment:
Should we make the signatures non-optional? The replica code MR currently declares the signatures as a non-optional field (same for error code).

Reply:
As far as I can tell this might be a good idea. Even if they are non-optional and populated old agents will simply ignore the additional field on deserialization. So there is no need for a rollout as an optional taking into account backward compatibility issues.

Reply:
Last time discussing about it, we think the optional field may be useful for boundary nodes if they fail to resemble enough matched query results. In this case, we'd want reject it at BN without node signatures.

However, it seems a plaintext http error is enough instead of a HttpQueryResponse::Rejected with node signatures missing.

Reply:
Thank you for your input! I've made the signatures field non-optional.